ambry-store/src/main/java/com.github.ambry.store/HardDeleter.java

```diff
@@ -111,17 +113,30 @@ public void run() {
     try {
       while (running.get()) {
         try {
-          if (!hardDelete()) {
-            isCaughtUp = true;
+          if (paused.get()) {
```

Comment:
Yes, I have cleaned this up and hopefully it looks good this time. If it could be simplified further, do let me know.